src/app/swap/swap-listener.ts

```diff
@@ -0,0 +1,61 @@
+import { toSats } from "@domain/bitcoin"
+
+import { admin as LedgerAdmin } from "@services/ledger/admin"
+import { WalletCurrency } from "@domain/shared"
+import { LedgerTransactionType } from "@domain/ledger"
+import { SwapState, SwapType } from "@domain/swap"
+import { addAttributesToCurrentSpan } from "@services/tracing"
+
+export const handleSwapOutCompleted = async (swapStatus: SwapStatusResultWrapper) => {
+  if (swapStatus.parsedSwapData) {
+    const type = swapStatus.parsedSwapData.swapType
+    const onchainMinerFee = toSats(swapStatus.parsedSwapData.onchainMinerFee)
+    const offchainRoutingFee = toSats(swapStatus.parsedSwapData.offchainRoutingFee)
+    const serviceProviderFee = toSats(swapStatus.parsedSwapData.serviceProviderFee)
+    const totalFees = onchainMinerFee + offchainRoutingFee + serviceProviderFee
+    const state = swapStatus.parsedSwapData.state
+    if (type === SwapType.SWAP_OUT && state === SwapState.SUCCESS && totalFees > 0) {
+      const swapFeeMetadata: SwapFeeLedgerMetadata = {
+        swapId: swapStatus.parsedSwapData.id,
+        swapAmount: toSats(swapStatus.parsedSwapData.amt),
+        htlcAddress: swapStatus.parsedSwapData.htlcAddress,
+        onchainMinerFee,
+        offchainRoutingFee,
+        serviceProviderFee,
+        serviceProvider: "LOOP",
+        currency: WalletCurrency.Btc,
+        type: LedgerTransactionType.Fee,
+        pending: false,
+      }
+      addAttributesToCurrentSpan({
+        "swap.success": JSON.stringify(swapFeeMetadata),
+      })
+      await recordSwapFeeToLedger(swapFeeMetadata)
+    }
+    if (state === SwapState.FAILED) {
+      addAttributesToCurrentSpan({
+        "swap.error": JSON.stringify({
+          swapiId: swapStatus.parsedSwapData.id,
+          message: swapStatus.parsedSwapData.message,
+          amt: swapStatus.parsedSwapData.amt.toString(),
+        }),
+      })
+    }
+  }
+  return true
+}
+
+// TODO handle
+export function handleSwapOutFailure(data) {
```

Comment:
are we not already handling the failure here?

https://github.com/GaloyMoney/galoy/pull/1396/files#diff-0e844e1434164c814e3fb21ae1cb0649254d367a58697369661b66a9587dd5b6R35

Reply:
one things that would be useful is to have an alert on loop out failure, but this would be outside this repo.